nomos-services/data-availability/sampling/src/lib.rs

```diff
@@ -115,15 +118,21 @@ where
                 sampler.handle_sampling_success(blob_id, *blob).await;
             }
             SamplingEvent::SamplingError { error } => {
+                // TODO: in most of these error cases we can't get the blob_id from the error
+                // Shouldn't the error contain that?
+                // We can of course stop tracking that blob_id in the backend via timeout,
+                // which we want to have anyways, but could it be nicer to remove it here too,
+                // by calling the handler_sampling_error method?
```

Comment:
This is a specific error. We can try to get the blob_id from the error. It is has we sort circuit the sampling (calling the error handling method with the blobid). Otherwise we wait till time expires. It can be done here or in a different PR.
Steps would be:
* Create a `blob_id` method for the sampling error which returns an `Option<BlobId>`
* Call the method here and check if it is `Some(blob_id)` to call the error handling method.
* Log it otherwise

Reply:
Added in this PR